src/redux/books/books.js

```diff
@@ -32,9 +29,55 @@ const booksReducer = (state = stateInit, action) => {
     case REMOVE_BOOK:
       return state.filter((book) => book.id !== action.payload);
 
+    case GET_BOOKS:
+      return action.payload;
+
     default:
       return state;
   }
 };
 
+export const getBooksList = () => async (dispatch) => {
+  const booksList = await fetch(`${url}`)
+    .then((response) => response.json());
+  const booksID = Object.keys(booksList);
+  const booksArr = [];
+  booksID.map((item) => booksArr.push({
+    id: item,
+    title: booksList[item][0].title,
+    author: booksList[item][0].author,
+    category: booksList[item][0].category,
+  }));
+  dispatch(getBooks(booksArr));
+};
+
+export const postBook = (newBook) => async (dispatch) => {
+  await fetch(`${url}`, {
+    method: 'POST',
+    body: JSON.stringify({
+      item_id: newBook.id,
+      title: newBook.title,
+      author: newBook.author,
+      category: newBook.category,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(addBook(newBook));
+};
+
+export const deleteBook = (id) => async (dispatch) => {
+  await fetch(`${url}/${id}`, {
+    method: 'DELETE',
+    body: JSON.stringify({
+      item_id: id,
+    }),
+    headers: {
+      'Content-type': 'application/json; charset=UTF-8',
+    },
+  });
+  dispatch(removeBook(id));
+};
+
```

Comment:
The almost same thing happens here: if I delete a book, it disappears from the list. But after refreshing, I see it back again. Unfortunately, the API is not informed about the changes.

- [ ] Kindly check what the issue may be regarding this aspect and when you are ready, please submit again. :+1: